corehq/messaging/scheduling/models/content.py

```diff
@@ -478,3 +479,42 @@ def send(self, recipient, logged_event, phone_entry=None):
             logged_subevent.error(MessagingEvent.ERROR_CANNOT_RENDER_MESSAGE, additional_error_text=str(error))
             raise
         logged_subevent.completed()
+
+
+class FCMNotificationContent(Content):
+    ACTION_CHOICES = [
+        ('SYNC', _('Background Sync'))
+    ]
+
+    MESSAGE_TYPE_NOTIFICATION = 'NOTIFICATION'
+    MESSAGE_TYPE_DATA = 'DATA'
+
+    MESSAGE_TYPES = [
+        (MESSAGE_TYPE_NOTIFICATION, _('Display Messages')),
+        (MESSAGE_TYPE_DATA, _('Data Messages'))
+    ]
+    # subject and message corresponds to 'title' and 'body' respectively in FCM terms.
```

Comment:
Could we use the same vocabulary here to avoid the overhead?

Reply:
Actually I am re-using the fields `subject` and `message` to keep consistent with the design used currently in Contents and Schedule Forms. This is a compromise made to avoid overhead of adding new form fields and keeping code consistent,